packages/forms/src/elements/common/Legend.tsx

```diff
@@ -0,0 +1,23 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import React, { forwardRef, HTMLAttributes } from 'react';
+import { StyledLegend } from '../../styled';
+import useFieldsetContext from '../../utils/useFieldsetContext';
+
+/**
+ * @extends HTMLAttributes<HTMLLegendElement>
+ */
+export const Legend = forwardRef<HTMLLegendElement, HTMLAttributes<HTMLLegendElement>>(
+  (props, ref) => {
+    const fieldsetContext = useFieldsetContext();
+
+    return <StyledLegend {...props} {...fieldsetContext} ref={ref} />;
```

Comment:
Should this be refined? With the recent change `isRegular` is being passed as an invalid prop to the `StyledLegend`.

Reply:
I think this should be okay. It looks like `StyledLegend` only consumes `isCompact`. Or should it also have an `isRegular` prop that matches the characteristics of `Label`?

Reply:
We don't want regular weight functionality applied to the `Legend`. Perhaps it would be more clear to destructure `isCompact` out for clarity. YMMV.